src/variable_rate.jl

```diff
@@ -387,8 +408,10 @@ function (cache::VR_DirectEventCache)(u, t, integrator)
     return cache.current_threshold
 end
 
-function execute_affect!(vjumps, integrator, idx)
-    @inbounds vjumps[idx].affect!(integrator)
+@generated function execute_affect!(cache::VR_DirectEventCache{T, F1, F2, RNG}, integrator, idx) where {T, F1, F2, RNG}
```

Comment:
```suggestion
@generated function execute_affect!(cache::VR_DirectEventCache{T, RNG, F1, F2}, integrator, idx) where {T, RNG, F1, F2 <: Tuple}
```
Again, you need to order these consistent with how `VR_DirectEventCache` is defined.

Reply:
Let's add the `Tuple` assertion on the type too as later we will have a different version of this function for vectors of function wrappers.